plugins/scaffolder-backend/src/scaffolder/stages/prepare/azure.test.ts

```diff
@@ -36,119 +29,69 @@ describe('AzurePreparer', () => {
+      'https://dev.azure.com/backstage-org/backstage-project/_git/template-repo',
+    workspacePath: '/tmp',
+    logger,
+  };
 
   it('calls the clone command with token from integrations config', async () => {
-    await preparer.prepare(mockEntity, { logger });
+    await preparer.prepare(prepareOptions);
 
     expect(Git.fromAuth).toHaveBeenCalledWith({
       logger,
       password: 'fake-azure-token',
       username: 'notempty',
     });
+    expect(fs.move).toHaveBeenCalledWith('/tmp/checkout', '/tmp/template');
+    expect(fs.rmdir).toHaveBeenCalledWith('/tmp/template/.git');
   });
 
   it('calls the clone command with the correct arguments for a repository', async () => {
-    await preparer.prepare(mockEntity, { logger: getVoidLogger() });
+    await preparer.prepare(prepareOptions);
 
     expect(mockGitClient.clone).toHaveBeenCalledWith({
       url:
         'https://dev.azure.com/backstage-org/backstage-project/_git/template-repo',
-      dir: expect.any(String),
+      dir: '/tmp/checkout',
     });
   });
 
   it('calls the clone command with the correct arguments for a repository when no path is provided', async () => {
-    delete mockEntity.spec.path;
-
-    await preparer.prepare(mockEntity, { logger: getVoidLogger() });
-
-    expect(mockGitClient.clone).toHaveBeenCalledWith({
+    await preparer.prepare({
       url:
         'https://dev.azure.com/backstage-org/backstage-project/_git/template-repo',
-      dir: expect.any(String),
+      workspacePath: '/tmp',
+      logger,
     });
-  });
-
-  it('return the temp directory with the path to the folder if it is specified', async () => {
-    mockEntity.spec.path = './template/test/1/2/3';
 
-    const response = await preparer.prepare(mockEntity, {
-      logger: getVoidLogger(),
+    expect(mockGitClient.clone).toHaveBeenCalledWith({
+      url:
+        'https://dev.azure.com/backstage-org/backstage-project/_git/template-repo',
+      dir: '/tmp/checkout',
     });
-
-    expect(response.split('\\').join('/')).toMatch(
-      /\/template\/test\/1\/2\/3$/,
-    );
+    expect(fs.move).toHaveBeenCalledWith('/tmp/checkout', '/tmp/template');
   });
 
-  it('return the working directory with the path to the folder if it is specified', async () => {
-    mockEntity.spec.path = './template/test/1/2/3';
-
-    const response = await preparer.prepare(mockEntity, {
-      workingDirectory: '/workDir',
-      logger: getVoidLogger(),
+  it('moves the template from path if it is specified', async () => {
+    const path = './template/test/1/2/3';
+    await preparer.prepare({
+      url: `https://dev.azure.com/backstage-org/backstage-project/_git/template-repo?path=${encodeURIComponent(
+        path,
+      )}`,
+      logger,
+      workspacePath: '/tmp',
     });
 
-    expect(response.split('\\').join('/')).toMatch(
-      /\/workDir\/graphql-starter-static\/template\/test\/1\/2\/3$/,
+    expect(fs.move).toHaveBeenCalledWith(
+      '/tmp/checkout/template/test/1/2/3',
```

Comment:
Tests on Windows have been mysterious with mounting file system at `/tmp`.

I think it's safer to initialize an OS aware root folder like

```ts
const tmpDir = os.platform() === 'win32' ? 'C:\\tmp' : '/tmp';
```

and then use `resolve` from `'path'` to construct a paths inside `tmpDir`.

A recent related PR https://github.com/backstage/backstage/pull/4249

Reply:
Great suggestion, thanks! 👍

Reply:
@OrkoHunter I made some modifications to the test, please take a look before I replicate it to the rest of the preparers.

Reply:
Your latest commit looks good to me @jhaals ! I have a trial of Windows VM via [parallels](https://www.parallels.com/) and so I took the liberty to run tests on this branch. Hope this log helps 😁 

https://pastebin.com/uQD1PG7a

I only ran the tests inside `plugins/scaffolder-backend`.

Reply:
Thanks a lot! I'm happy to see ` PASS  src/scaffolder/stages/prepare/azure.test.ts` :)